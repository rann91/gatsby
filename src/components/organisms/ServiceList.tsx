/* @jsx jsx */
import { Container, jsx, Flex, Box } from 'theme-ui'
import {
  Strapi_ComponentContentServiceList,
  Strapi_ComponentSharedService
} from '../../typings/graphql'
import Section from '../atoms/Section'
import HeadingGroup from '../molecules/HeadingGroup'
import { column } from '../../gatsby-plugin-theme-ui'
import Service from '../molecules/Service'

const ServiceList = ({
  title,
  subtitle,
  services
}: Strapi_ComponentContentServiceList) => (
  <Section data-testid="service-list" hasTitle={!!title} sx={{ mb: 5 }}>
    <Container>
      <HeadingGroup {...{ title, subtitle }} />
      <Flex
        sx={{
          flexWrap: 'wrap',
          '* + &': {
            paddingTop: [0, null, null, 5]
          }
        }}>
        {((services || []) as Strapi_ComponentSharedService[]).map(service => (
          <Box
            key={service.id}
            p={5}
            sx={{
              width: [column(12), null, column(6), column(4)],
              textAlign: ['center', null, null, 'left']
            }}>
            <Service {...service} />
          </Box>
        ))}
      </Flex>
    </Container>
  </Section>
)

export default ServiceList
