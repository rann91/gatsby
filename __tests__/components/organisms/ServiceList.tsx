import { render } from '@testing-library/react'
import React from 'react'
import ServiceList from '../../../src/components/organisms/ServiceList'
import component from '../../fixtures/component-service-list.json'

describe('ServiceList', () => {
  it('renders correctly', () => {
    const { container } = render(<ServiceList {...component} />)

    expect(container).toMatchSnapshot()
  })
})
