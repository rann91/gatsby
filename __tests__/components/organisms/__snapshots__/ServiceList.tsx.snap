// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceList renders correctly 1`] = `
<div>
  <section
    class="css-10nqdfk-ServiceList"
    data-testid="service-list"
  >
    <div
      class="css-1yrmf69"
    >
      <div
        class="css-1mpaekt-HeadingGroup"
      >
        <div
          class="css-146vepc-HeadingGroup"
          data-testid="subtitle"
        >
          What we are best at
        </div>
        <h2
          class="css-16nj6cn-HeadingGroup"
          data-testid="title"
        >
          Our services
        </h2>
      </div>
      <div
        class="css-1x8he57-ServiceList"
      >
        <div
          class="css-13dda3r"
        >
          <div
            class="css-5uwokq-Service"
          >
            <div
              class="css-i9l3dq"
            >
              <div
                class="css-tmeqbf"
              />
              <div
                class="css-ojfau8"
              >
                <img
                  alt="Service logo"
                  class="css-1vz0lri-Service"
                  src="/static/e1f5c103db11513ce0e1488bf71a12c7/support_734bde873c.svg"
                />
              </div>
            </div>
          </div>
          <h3
            class="css-j6wujt-Service"
          >
            Support
          </h3>
          <p
            class="css-vurnku"
          >
            Far far away, behind the word mountains, far from the countries Vokalia and Consonantia, there live the blind texts.
          </p>
        </div>
        <div
          class="css-13dda3r"
        >
          <div
            class="css-5uwokq-Service"
          >
            <div
              class="css-i9l3dq"
            >
              <div
                class="css-tmeqbf"
              />
              <div
                class="css-ojfau8"
              >
                <img
                  alt="Service logo"
                  class="css-1vz0lri-Service"
                  src="/static/03ccc4247828dd9f72de92019e7fbe45/strategy_4a02bf0bd6.svg"
                />
              </div>
            </div>
          </div>
          <h3
            class="css-j6wujt-Service"
          >
            Strategy
          </h3>
          <p
            class="css-vurnku"
          >
            Even the all-powerful Pointing has no control about the blind texts it is an almost unorthographic life.
          </p>
        </div>
        <div
          class="css-13dda3r"
        >
          <div
            class="css-5uwokq-Service"
          >
            <div
              class="css-i9l3dq"
            >
              <div
                class="css-tmeqbf"
              />
              <div
                class="css-ojfau8"
              >
                <img
                  alt="Service logo"
                  class="css-1vz0lri-Service"
                  src="/static/bad3bd60a3d599f881f0e74d19c2f43f/management_7eca47eb5b.svg"
                />
              </div>
            </div>
          </div>
          <h3
            class="css-j6wujt-Service"
          >
            Management
          </h3>
          <p
            class="css-vurnku"
          >
            Blind texts it is an almost unorthographic life One day however a small line of blind text by the name.
          </p>
        </div>
        <div
          class="css-13dda3r"
        >
          <div
            class="css-5uwokq-Service"
          >
            <div
              class="css-i9l3dq"
            >
              <div
                class="css-tmeqbf"
              />
              <div
                class="css-ojfau8"
              >
                <img
                  alt="Service logo"
                  class="css-1vz0lri-Service"
                  src="/static/1d262bcb471c28b6d378459d8ee7bd55/training_c092d67f1d.svg"
                />
              </div>
            </div>
          </div>
          <h3
            class="css-j6wujt-Service"
          >
            Training
          </h3>
          <p
            class="css-vurnku"
          >
            Blind texts it is an almost unorthographic life One day however a small line of blind text by the name.
          </p>
        </div>
        <div
          class="css-13dda3r"
        >
          <div
            class="css-5uwokq-Service"
          >
            <div
              class="css-i9l3dq"
            >
              <div
                class="css-tmeqbf"
              />
              <div
                class="css-ojfau8"
              >
                <img
                  alt="Service logo"
                  class="css-1vz0lri-Service"
                  src="/static/d49c9a3ec392fe1419264e4a74f119cf/consulting_3db53fea72.svg"
                />
              </div>
            </div>
          </div>
          <h3
            class="css-j6wujt-Service"
          >
            Consulting
          </h3>
          <p
            class="css-vurnku"
          >
            Far far away, behind the word mountains, far from the countries Vokalia and Consonantia, there live the blind texts.
          </p>
        </div>
        <div
          class="css-13dda3r"
        >
          <div
            class="css-5uwokq-Service"
          >
            <div
              class="css-i9l3dq"
            >
              <div
                class="css-tmeqbf"
              />
              <div
                class="css-ojfau8"
              >
                <img
                  alt="Service logo"
                  class="css-1vz0lri-Service"
                  src="/static/98dfe375049aea2c3b10efa8dafd9364/events_171bfcf6ed.svg"
                />
              </div>
            </div>
          </div>
          <h3
            class="css-j6wujt-Service"
          >
            Events
          </h3>
          <p
            class="css-vurnku"
          >
            Even the all-powerful Pointing has no control about the blind texts it is an almost unorthographic life.
          </p>
        </div>
      </div>
    </div>
  </section>
</div>
`;
